### PR TITLE
Add extra swarm lock/unlock tests that ensure that all the managers are locked with the same key

### PR DESCRIPTION
**Pull request by cyli**
cc @aaronlehmann 

The tests would be useful to add for 1.13.

### CODE REVIEW

File: integration-cli/docker_cli_swarm_test.go [hunk 1]
```diff
@@ -946,10 +951,164 @@ func (s *DockerSwarmSuite) TestSwarmLeaveLocked(c *check.C) {
...
+
+		cmd := d.command("swarm", "unlock")
+		cmd.Stdin = bytes.NewBufferString(unlockKey)
+		out, err := cmd.CombinedOutput()
+		c.Assert(err, checker.IsNil, check.Commentf("out: %v", string(out)))
+		checkLockStatus(c, d, false)
+	}
+
+	// d2 never got the cluster update, so it is still set to unlocked
+	c.Assert(d2.Restart(), checker.IsNil)
```
Comment (aaronlehmann) [inline integration-cli/docker_cli_swarm_test.go]: `d2.Start`, since it is already stopped

File: integration-cli/docker_cli_swarm_test.go [hunk 2]
```diff
@@ -946,10 +951,164 @@ func (s *DockerSwarmSuite) TestSwarmLeaveLocked(c *check.C) {
...
+		}
+	}
+
+	c.Assert(unlockKey, checker.Not(checker.Equals), "")
+
+	outs, err = d1.Cmd("swarm", "unlock-key", "-q")
+	c.Assert(outs, checker.Equals, unlockKey+"\n")
+
+	// wait for everything to be locked
+	time.Sleep(1 * time.Second)
```
Comment (aaronlehmann) [inline integration-cli/docker_cli_swarm_test.go]: I'm afraid the test will be fragile with these sleeps. Can we copy the strategy in `TestSwarmRotateUnlockKey` where we retry with a longer sleep if we didn't get the expected result?

File: integration-cli/docker_cli_swarm_test.go [hunk 3]
```diff
@@ -946,10 +951,164 @@ func (s *DockerSwarmSuite) TestSwarmLeaveLocked(c *check.C) {
...
+	columns := strings.Fields(lines[len(lines)-1])
+	c.Assert(columns, checker.HasLen, 4) // if it was a manager it'd have a manager status field
+
+	c.Assert(d3.Restart(), checker.IsNil)
+	checkLockStatus(c, d3, false)
+}
+
 func (s *DockerSwarmSuite) TestSwarmRotateUnlockKey(c *check.C) {
-	d := s.AddDaemon(c, true, true)
+	d1 := s.AddDaemon(c, true, true) // leader - don't restart this one, we don't want leader election delays
```
Comment (aaronlehmann) [inline integration-cli/docker_cli_swarm_test.go]: On second thought, I'd rather add this as a separate test in addition to the existing `TestSwarmRotateUnlockKey`. Not restarting `d1` does change what we're testing a bit. The original test was really good at finding problems.

**Comment by cyli** [comment]
2 of the new tests (https://github.com/docker/docker/pull/28566/files#diff-bea1a591876503439a06ad068ca3ae8fR1226) could be flakey since sometimes they trigger https://github.com/docker/swarmkit/pull/1766.  Once https://github.com/docker/swarmkit/pull/1766 is in that should go away.

**Comment by aaronlehmann** [comment]
LGTM

**Comment by aaronlehmann** [comment]
Let's try to merge these before #28907

cc @tonistiigi

**Comment by tonistiigi** [comment]
LGTM

**Comment by thaJeztah** [comment]
@vieux LMK if you want this cherry-picked into the 1.13 branch

**Pull request merged**